src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MINIMAL_GROWING_STEP = 1;
+    private static final int MINIMAL_SIZE_LIST = 0;
+    private static final int START_INDEX = 0;
+    private static final double MINIMAL_GROWING_RANGE = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void rangeCheckForAdd(int index) {
+        if (index >= size || index < MINIMAL_SIZE_LIST) {
+            throw new ArrayListIndexOutOfBoundsException("No such index: " + index);
+        }
+    }
+
+    private void ensureCapacity(int minCapacity) {
+        if (minCapacity > elementData.length) {
+            T[] newArray = (T[]) new Object[Math.max(elementData.length
+                    * (int) MINIMAL_GROWING_RANGE, minCapacity)];
```

Comment:
Not a change request, since the tests passed. But, aren't cast to (int) make MINIMAL_GROWING_RANGE = 1?

Reply:
Yes, it was my mistake. I wanted to cast Math